apps/prairielearn/assets/scripts/instructorQuestionSettingsClient.ts

```diff
@@ -11,6 +19,46 @@ onDocumentReady(() => {
   );
   const saveButton = document.querySelector<HTMLButtonElement>('#save-button');
 
+  new TomSelect('#topic', {
+    valueField: 'name',
+    searchField: ['name'],
+    closeAfterSelect: true,
+    items: [(document.querySelector('#topic') as HTMLSelectElement).dataset.selectedTopic],
+    plugins: ['dropdown_input'],
+    maxItems: 1,
+    placeholder: 'Select a topic',
+    render: {
+      option(data: TopicData, escape: (input: string) => string) {
+        return (
+          '<div>' +
+          '<span class="badge justify-content-start color-' +
+          escape(data.color) +
+          '">' +
+          escape(data.name) +
+          '</span>' +
+          '<div class="w-100 d-flex">' +
+          '<small class="text-muted justify-content-start text-start">' +
+          escape(data.description) +
+          '</small>' +
+          '</div>' +
+          '</div>'
+        );
+      },
+      item(data: TopicData, escape: (input: string) => string) {
+        return (
+          '<div class="w-100 d-flex justify-content-between align-items-center">' +
+          '<div class="btn btn-ghost badge color-' +
+          escape(data.color) +
+          '">' +
+          escape(data.name) +
+          '</div>' +
+          '<i class="fas fa-angle-down mx-2"></i>' +
+          '</div>'
```

Comment:
Produces an input that looks like this: ![Screenshot 2024-11-20 at 1 42 10 PM](https://github.com/user-attachments/assets/0a658c1b-8cc5-4b1b-a140-ee1ee94770d3)

Note, because `tom-select` replaces the original `<select>` element with a `<div>`, there by default is no dropdown arrow in the element. In [16ae108](https://github.com/PrairieLearn/PrairieLearn/pull/10868/commits/16ae10840b04823c6fa0ba356f87efa3828ca7d6), I added an icon that looks like what you would expect in a `<select>` element (i.e., a down arrow or downward caret). I had tried making this render like a normal `<select>` through Bootstrap classes but that resulted in a strange double border box. If the icon is incorrect or there is a better way to do this, let me know.